source/MaterialXRuntime/Commands/PrimCommands.cpp

```diff
@@ -35,6 +35,27 @@ void createPrim(RtStagePtr stage, const RtToken& typeName, const RtPath& path, R
 
 void createPrim(RtStagePtr stage, const RtToken& typeName, const RtPath& parentPath, const RtToken& name, RtCommandResult& result)
```

Comment:
Adds validation to the createPrim command.

Reply:
I think it would be better to move the validation logic to inside the `PvtCreatePrimCmd` class itself.  Since this function is just a wrapper to simplify creation and execution of the command.

That way we have all logic in a single place. And validation will always done no matter how the command is executed.

Reply:
Also see below comment on adding validation to each schema's createPrim function. It could be that is all that is needed and no logic is needed in the create command class.

Reply:
Actually we decided to move things to entirely different locations. Take another peek. The biggest thing I'm unsure of is where the function: throwIfNotTopLevelElement should live. Right now it's in Look.h, not sure that makes sense though.

Reply:
Actually, hold off a little bit, just realized I have to remove some code I forgot to and retest first, thanks.

Reply:
Ok, you guys should be able to look at this again.